keras/engine/training.py

```diff
@@ -888,10 +822,13 @@ def compile(self, optimizer, loss, metrics=None, loss_weights=None,
             else:
                 shape = self.internal_output_shapes[i]
                 name = self.output_names[i]
-                target = K.placeholder(ndim=len(shape),
-                                       name=name + '_target',
-                                       sparse=K.is_sparse(self.outputs[i]),
-                                       dtype=K.dtype(self.outputs[i]))
+                if K.is_placeholder(self.outputs[i]):
+                    target = K.placeholder(ndim=len(shape),
```

Comment:
Problem solved. The way I achieved this was to add a new labels parameter to the constructor of Container (and thus Model. So now there is an option to call `Model(inputs=inputs, outputs=outputs, labels=labels)`.